firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,118 @@
+                time.sleep(1)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_not_badware_button(label, report_page)
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "get me out" button
+                self.marionette.navigate(unsafe_page)
+                # Wait for the DOM to receive events for about:blocked
+                time.sleep(1)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_get_me_out_of_here_button()
+
+                # Disabled until bug 1084289 can be resolved
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "X" button
+                # self.marionette.navigate(unsafe_page)
+                # Wait for the DOM to receive events for about:blocked
+                # time.sleep(1)
+                # self.check_ignore_warning_button(unsafe_page)
+                # self.check_x_button()
+
+    def check_ignore_warning_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, 'ignoreWarningButton')
+        button.click()
+
+        self.wait_for_condition(expected.element_present(By.ID, 'main-feature'))
+        self.wait_for_condition(expected.element_stale(button))
+        self.assertEquals(self.marionette.get_url(), self.browser.get_final_url(unsafe_page))
+
+        # Clean up here since the permission gets set in this function
+        self.utils.remove_perms('www.itisatrap.org', 'safe-browsing')
+
+    # Check the not a forgery or attack button in the notification bar
+    def check_not_badware_button(self, button_property, report_page):
+        with self.marionette.using_context('chrome'):
+            # TODO: update to use safe browsing notification bar class when bug 1139544 lands
+            label = self.browser.get_property(button_property)
+            button = (self.marionette.find_element(By.ID, 'content')
+                      .find_element('anon attribute', {'label': label}))
+            # TODO: better workaround for timeout exceptions (especially stopbadware.org)
+            self.browser.tabbar.open_tab(lambda _: button.click())
+            self.wait_for_condition(lambda _: report_page in self.browser.navbar.locationbar.value)
+            self.browser.tabbar.close_tab()
+
+    def check_get_me_out_of_here_button(self):
+        with self.marionette.using_context('chrome'):
+            # TODO: update to use safe browsing notification bar class when bug 1139544 lands
+            label = self.browser.get_property('safebrowsing.getMeOutOfHereButton.label')
+            button = (self.marionette.find_element(By.ID, 'content')
+                      .find_element('anon attribute', {'label': label}))
+            button.click()
+
+        self.wait_for_condition(lambda mn: self.browser.default_homepage in mn.get_url())
```

Comment:
Good idea! That way we do not have to wait until the final page has been loaded.